README.md

```diff
@@ -1,2 +1,218 @@
+
+*Default value:* `''`
+
+**`distance`**
+
+A function to calculate the distance between two equal-dimension vectors, used in calculating the distance matrix, in the format:
+
+```javascript
+function (arrayA, arrayB) { return someNumber; }
+```
+
+The function receives two equal-length arrays of numbers.
+
+*Default value:* `euclideanDistance` from this `hclust` package
+
+**`linkage`**
+
+A function to calculate the distance between pairs of clusters, used in determining linkage criterion, in the format:
+
+```javascript
+function (arrayA, arrayB, distanceMatrix) { return someNumber; }
+```
+
+The function receives two sets of indexes and the distance matrix computed between each datum and every other datum.
+
+*Default value:* `averageDistance` from this `hclust` package
+*Other built-in values:* `minDistance` and `maxDistance` from this `hclust` package
+
+**`onProgress`**
+
+A function that is called several times throughout clustering, and is provided the current progress through the clustering, in the format:
+
+```javascript
+function (progress) { }
+```
+
+The function receives the percent progress between `0` and `1`.
+
+*Default value:* an internal function that `console.log`'s the progress
+
+**Note:** [`postMessage`](https://developer.mozilla.org/en-US/docs/Web/API/Worker/postMessage) is called in the same places as `onProgress`, if the script is running as a [web worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers).
+
+#### Returns
+
+**`clusters`**
+
+The resulting cluster tree, in the format:
+
+```javascript
+{
+  indexes: [ ... Number, Number, Number ... ],
+  height: Number,
+  children: [ ... {}, {}, {} ... ]
+}
+
+```
+
+**`distances`**
+
+The computed distance matrix, in the format:
+
+```javascript
+[
+  ...
+  [ ... Number, Number, Number ...],
+  [ ... Number, Number, Number ...],
+  [ ... Number, Number, Number ...]
+  ...
+]
+```
+
+**`order`**
+
+The new order of the data, in terms of original data array indexes, in the format:
+
+```javascript
+[ ... Number, Number, Number ... ]
+```
+
+Equivalent to `clusters.indexes` and `clustersGivenK[1]`.
+
+**`clustersGivenK`** 
+
+A list of tree slices in terms of original data array indexes, where index = K, in the format:
+
+```javascript
+[
+  [], // K = 0
+  [ [] ], // K = 1
+  [ [], [] ], // K = 2
+  [ [], [], [] ], // K = 3
+  [ [], [], [], [] ], // K = 4
+  [ [], [], [], [], [] ] // K = 5
+  ...
+]
+```
+
+---
+
+### `avgDistance(arrayA, arrayB, distanceMatrix)`
+
+Calculates the average distance between pairs of clusters.
+
+---
+
+### `minDistance(arrayA, arrayB, distanceMatrix)`
+
+Calculates the smallest distance between pairs of clusters.
+
+---
+
+### `maxDistance(arrayA, arrayB, distanceMatrix)`
+
+Calculates the largest distance between pairs of clusters.
+
+---
+
+### Comparison with [hcluster.js](https://github.com/cmpolis/hcluster.js)
+
+- This package does not include items from the original dataset in the results.
```

Comment:
Will change to "does not duplicate items from the original data set" to make it clearer why I did it this way (to avoid data duplication).